### PR TITLE
test: make it easier to TDD integration tests

### PR DESCRIPTION
**Pull request by lundeeell**
**Ticket:** [#54](https://trello.com/c/2AeoXSfP/54-make-it-easier-to-tdd-with-integration-tests)

**Intent:**
Make it easier to TDD integration tests without using integration-test.sh

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,10 +83,15 @@ To overide default environment variables add an `.env` file in root.
...
 
-To run the integration-tests
+**To run the integration-tests once**
 
 1. Run `sh integration-tests.sh`
 
+**To develop with integration tests**
+
+1. Run dev docker-compose: `docker-compose -f docker-compose.dev.yml up -d`
+2. Run integration tests, i.g.: `docker-compose -f docker-compose.dev.yml exec -T myskills-api npm run integration -- --watchAll`
```
Comment (Graborg) [inline README.md]:
```suggestion
2. Run integration tests, e.g.: `docker-compose -f docker-compose.dev.yml exec -T myskills-api npm run integration -- --watchAll`
```

Comment (lundeeell) [reply README.md]: Oh, I actually meant e.g. The user can add other flags to jest, like test regex etc. Will fix!

File: test/integration/integrationUtils.ts [hunk 2]
```diff
@@ -1,17 +1,15 @@
 import got from 'got'
-
-const MYDATA_APP_URL = process.env.MYDATA_APP
 import { consents } from '../../lib/adapters/mydata'
 import { Login } from '../../lib/__generated__/myskills'
 import { defaultRequest } from '../../lib/services/consents'
 import { getConsentRequest } from '../../lib/services/db'
+import { createTestClient } from 'apollo-server-testing'
 
+const MYDATA_APP_URL = process.env.MYDATA_APP || 'http://127.0.0.1:1337'
```
Comment (Graborg) [inline test/integration/integrationUtils.ts]: I think fallbacks like this will only create confusion.

File: test/integration/setup.ts [hunk 3]
```diff
@@ -7,4 +7,5 @@ jest.unmock('../../lib/adapters/pubsub')
 
 // Randomize server port in order to run integration tests concurrent
 process.env.SERVER_PORT = `${Math.floor(Math.random() * 999) + 3000}`
-process.env.DOMAIN = `http://myskills-api:${process.env.SERVER_PORT}`
+process.env.DOMAIN = `http://${process.env.DOMAIN_HOST ||
```
Comment (Graborg) [inline test/integration/setup.ts]: Maybe it is time to remove 'host.docker.internal' and let people set it themselves if they want to run it outside of docker?

File: docker-compose.dev.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+services:
+  myskills-api:
+    build:
+      context: .
+      target: base
+    volumes:
+      - ./lib:/app/lib
+      - ./test:/app/test
+      - ./.git:/app/.git
```
Comment (Graborg) [inline docker-compose.dev.yml]: what is git needed for?

Comment (Graborg) [reply docker-compose.dev.yml]: Also, should this list not also include ".env". So as to let the user change it more easily? Although I think env variables have precidence over env. If that is so, then we might want to reconsider...

Comment (lundeeell) [reply docker-compose.dev.yml]: Git is needed if you want to run jest --watch
Dont think that envs was needed, as you can edit easily in docker-compose, but can add that

File: README.md [hunk 5]
```diff
@@ -2,10 +2,23 @@
...
+### Running in container
+
+The easiest way to develop is by running everything using the dev docker-compose:
+
+- Clone [myskills-api](https://github.com/JobtechSwe/myskills-api/) repo
+- Create local docker network: `docker network create myskills`
+- Run the docker-services: `docker-compose -f docker-compose.dev.yml up -d`
+- Go to `localhost:3000/graphql`
+
+Alternative, run node on local:
```
[the remark this answers is not in the document]
Comment (lundeeell) [reply README.md]: Good point

File: test/integration/setup.ts (hunk 3 above)
Comment (lundeeell) [reply test/integration/setup.ts]: Well, that can be another PR if so. But I think it is nice if it works both ways

File: docker-compose.dev.yml (hunk 4 above)
Comment (Graborg) [reply docker-compose.dev.yml]: Aha, cool! Lets keep it then

File: test/integration/integrationUtils.ts (hunk 2 above)
Comment (lundeeell) [reply test/integration/integrationUtils.ts]: Hmm... i disagree for now 👍

**Comment by Graborg** [comment]
A nice improvement. Although I did had to to add `--build` for running the docker compose, since new npm packages were added since the last time the image was built, and the old ones were still cached. It might be nice to point that out for people as well, still this is approved!

**Pull request merged**